R/fit.R

```diff
@@ -415,12 +418,21 @@ mmrm <- function(formula,
     drop_visit_levels = control$drop_visit_levels,
     allow_na_response = FALSE
   )
-  fit <- fit_single_optimizer(
-    tmb_data = tmb_data,
-    formula_parts = formula_parts,
-    control = control
-  )
-
+  fit <- list()
+  while ((length(fit) == 0) && length(control$optimizers) > 0) {
+    fit <- fit_single_optimizer(
+      tmb_data = tmb_data,
+      formula_parts = formula_parts,
+      control = control
+    )
+    if (length(fit) == 0) {
+      warning(paste0(
```

Comment:
I wonder if we really need this warning (because other divergent issues might be captured by the optimizers, like different convergence code; and other non-converged issue happens we just silently go to other optimizers).  Maybe also good to capture all the warning/errors in `do.call` part of the optimizer so that anyway we have a returned fit in `fit_mmrm` but all the errors/logs are stored. (`tmb_opt` result can be manually constructed with `list(value = NA, par = initial values etc, convergence = 1 etc, error = caught error)`

Reply:
Good ideas! I thought about this a bit more after our stand up.

- I think I would not move the capturing of errors and warnings further inside, because `fit_mmrm` (one level down) is supposed to be low level, i.e. not capture these. (and if we would consider this, it would be quite big change with separate issue/PR)
- So far all the examples we have seen recently about these divergence errors/warnings were actually pointing towards overparametrized covariance models (instead of just unfortunate optimizer behavior). Therefore I think the warning makes sense to alert the user, while still trying other optimizers.